backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/CalendarsService.cs

```diff
@@ -0,0 +1,150 @@
+            return true;
+        }
+
+        public async Task<bool> UpdateGoogleCalendar(List<UserCalendarDto> calendarDtoList)
+        {
+            foreach (var calendarDto in calendarDtoList)
+            {
+                var calendar = await _context.Calendars
+                    .Include(c => c.ImportEventsFromTeam)
+                    .Include(c => c.VisibleForTeams)
+                    .FirstOrDefaultAsync(el => el.Id == calendarDto.Id);
+
+                _context.CalendarVisibleForTeams.RemoveRange(calendar!.VisibleForTeams);
+                calendar.VisibleForTeams = Array.Empty<CalendarVisibleForTeam>();
+
+                var newVisibleForList = calendarDto.VisibleForTeams?
+                    .Select(el => new CalendarVisibleForTeam
+                    {
+                        CalendarId = calendar.Id,
+                        TeamId = el.Id,
+                        IsDeleted = false,
+                    }).ToList();
+
+                if (newVisibleForList != null)
+                {
+                    calendar.VisibleForTeams = newVisibleForList;
+                    await _context.CalendarVisibleForTeams.AddRangeAsync(newVisibleForList);
+                }
+
+                calendar.CheckForConflicts = calendarDto.CheckForConflicts;
+                calendar.AddEventsFromTeamId = calendarDto.ImportEventsFromTeam?.Id;
+            
+                _context.Calendars.Update(calendar);
+                await _context.SaveChangesAsync();
+            }
+            
+            return true;
+        }
+
+        public async Task<List<UserCalendarDto>> GetCurrentUserCalendars()
+        {
+            var currentUserEmail = GetCurrentUserEmail();
+            var currentUser = await _context.Users.FirstAsync(el => el.Email == currentUserEmail);
+            
+            var calendarsList = await _context.Calendars
+                .Where(c => c.UserId == currentUser.Id)
+                .Include(c => c.User)
+                .Include(c => c.ImportEventsFromTeam)
+                .Include(c => c.VisibleForTeams)
+                    .ThenInclude(v => v.Team)
+                .ToListAsync();
+
+            return _mapper.Map<List<UserCalendarDto>>(calendarsList);
+        }
+
+        public async Task<bool> DeleteCalendar(long id)
+        {
+            var calendar = await _context.Calendars
+                .Include(el => el.VisibleForTeams)
+                .FirstOrDefaultAsync(el => el.Id == id);
+
+            if (calendar != null)
+            {
+                _context.Calendars.Remove(calendar);
+                await _context.SaveChangesAsync();
+                return true;
+            }
+
+            return false;
+        }
+
+        private string GetCurrentUserEmail()
+        {
+            var claimsList = _httpContextAccessor.HttpContext!.User.Claims.ToList();
+            var email = claimsList.Find(el => el.Type == ClaimTypes.Email);
+            return email!.Value;
+        }
+    }
```

Comment:
And it would be better to use Id instead of email

Reply:
will be it okay if I move this method (with getting user id instead email) to BaseController ? Because all our controllers protected with Authorize and it could be better to have this method in our common controllers class